src/sagas/projects.js

```diff
@@ -47,6 +50,17 @@ export function* changeCurrentProject() {
   yield call(saveCurrentProject, state);
 }
 
+export function* rehydrateProject({payload: {rehydratedProject}}) {
+  const state = yield select();
+  const allProjectKeys = getProjectKeys(state);
+  if (allProjectKeys.includes(rehydratedProject.projectKey)) {
+    yield put(__changeCurrentProject(rehydratedProject.projectKey));
+  } else {
+    yield put(projectLoaded(rehydratedProject));
+    yield put(__changeCurrentProject(rehydratedProject.projectKey));
```

Comment:
Here's my understanding of your suggestion:

`rehydrateProject()` dispatches `REHYDRATED_PROJECT_IMPORTED` by calling `projectRestoredFromLastSession()`. If you're *not* authenticated, the reducer creates a new project from the rehydrated version. If you *are* authenticated, the reducer loads the saved project from Firebase and overwrites it with the rehydrated version.

Is that what you had in mind? Sounds reasonable to me! Let me know if I've misunderstood.